### PR TITLE
Revert "Resolving automatic update edge cases (#3142)"

### PR DESCRIPTION
**Pull request by quanganhdo**
Task/Issue URL:
Tech Design URL:
CC:

**Description**:

**Steps to test this PR**:
1.

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

**Comment by quanganhdo** [comment]
* **#3274** <a href="https://app.graphite.dev/github/pr/duckduckgo/macos-browser/3274?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @quanganhdo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: DuckDuckGo/Preferences/Model/AboutPreferences.swift [hunk 1]
```diff
@@ -27,28 +27,25 @@ final class AboutPreferences: ObservableObject, PreferencesTabOpening {
 #if SPARKLE
     enum UpdateState {
 
-        case loading
         case upToDate
-        case newVersionAvailable
+        case updateCycle(UpdateCycleProgress)
```
Comment (tomasstrba) [inline DuckDuckGo/Preferences/Model/AboutPreferences.swift:31]: Nice 👍

File: DuckDuckGo/Updates/UpdateController.swift [hunk 2]
```diff
@@ -151,80 +142,68 @@ final class UpdateController: NSObject, UpdateControllerProtocol {
         }
     }
 
-    func checkForUpdate() {
+    func checkForUpdateIfNeeded() {
+        guard let updater, !updater.sessionInProgress else { return }
+
         Logger.updates.debug("Checking for updates")
```
Comment (tomasstrba) [inline DuckDuckGo/Updates/UpdateController.swift:148]: There was a mistake when refactoring pixels and all update pixels are just debug pixels. Do you think it would be beneficial to make them available in release builds for potential debugging?

**Review by tomasstrba — COMMENTED** [review]
The code looks really good. 💯 I'm going to test the solution now

**Review by tomasstrba — COMMENTED** [review]
@quanganhdo, after installation of the build number 261, it says the app is up to date which is not expected. Am I missing something? I'm currently at the coffee shop with a low connection or maybe the phased rollout is the reason?
Screenshot 2024-09-18 at 09 53 10

Another minor thing is the empty box in the release notes which I suppose is caused by empty release notes in the appcast. Just flagging it in case not.

**Comment by quanganhdo** [comment]
Was that the review build I attached in the task? It should've pointed to my test appcast: https://github.com/quanganhdo/playground/blob/main/appcast2.xml

File: DuckDuckGo/Updates/UpdateController.swift (hunk 2 above)
Comment (quanganhdo) [reply DuckDuckGo/Updates/UpdateController.swift:148]: It may be beneficial to have pixels at different states to see if our users make it to the end of the update cycle. Have we defined those, or do they have to be created?

Comment (tomasstrba) [reply DuckDuckGo/Updates/UpdateController.swift:148]: There are couple of pixels and the error pixel in case something goes wrong. Monitoring various states as you suggested may be a good idea too 👍

**Review by tomasstrba — APPROVED** [review]
LGMT! 👍 Works as expected including edge cases

**Review by tomasstrba — APPROVED** [review]
Final LGTM! ✅ 

@quanganhdo, good job! 👏  This is a great improvement, I appreciate you dived into it and found a better solution. 🏅 

As we discussed during our 1:1 today, one final test with review builds would be awesome, but it's not required since we'll be testing it during the internal release phase next week
